File: src/main/java/leets/leenk/domain/notification/application/dto/response/NotificationResponse.java
package leets.leenk.domain.notification.application.dto.response;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonInclude;

import leets.leenk.domain.notification.domain.entity.NotificationContent;
import leets.leenk.domain.notification.domain.entity.enums.NotificationType;
import lombok.Builder;

@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public record NotificationResponse(
        String id,
        Long userId,
        NotificationType notificationType,
        String path,
        Boolean isRead,
        NotificationContent content,
        LocalDateTime updateDate
) {
}

File: src/main/kotlin/leets/leenk/domain/notification/application/dto/response/NotificationResponse.kt
package leets.leenk.domain.notification.application.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude
import leets.leenk.domain.notification.domain.entity.NotificationPayload
import leets.leenk.domain.notification.domain.entity.enums.NotificationType
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class NotificationResponse(
    val id: String,
    val userId: Long,
    val notificationType: NotificationType,
    val path: String,
    val isRead: Boolean,
    val content: NotificationPayload,
    val updateDate: LocalDateTime,
)
